tpm2tools/flush_active_handles.go

```diff
@@ -0,0 +1,40 @@
+package tpm2tools
+
+
+import (
+    "io"
+    "github.com/google/go-tpm/tpm2"
+    "github.com/google/go-tpm/tpmutil"
+)
+
+// FlushActiveHandles calls FlushContext() on all active handles within the 
+// TPM at io.ReadWriter rw. Returns nil if successful or TPM has no active
+// handles.
+func FlushActiveHandles(rw io.ReadWriter) error {
+    handles, err := activeHandles(rw)
+    if err != nil {
+        return err
+    }
+    for _, handle := range handles {
+        err = tpm2.FlushContext(rw, handle)
+        if err != nil {
+            return err
+        }
+    }
+    return nil
+}
+
+// activeHandles returns a slice of tpmutil.Handle objects of all handles
```

Comment:
In general, I think this method should be public and be parameterized over the _type_ of handles we want to list. This method assumes that we want to list the transient handles (which is correct if we are trying to clear the handles). However, I would like it better if some sort of `HandleType` was defined and had constants corresponding to values of TPM_HT. [See table 2:27 in part 4 of the spec](https://trustedcomputinggroup.org/wp-content/uploads/TPM-Rev-2.0-Part-4-Supporting-Routines-01.38-code.pdf)

The name could be something like ListHandles (or similar)

Reply:
Done.